### PR TITLE
Added _util directory.

### PR DESCRIPTION
**Pull request by Thom1729**
~~These utilities should have tests.~~

`glob_util` is probably a good candidate for this directory. Probably also some of the stuff from `settings_dict` and `syntax` (roughly, the stuff that's dragging down the coverage).

### CODE REVIEW

File: st3/sublime_lib/_util/collection_util.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
+from collections.abc import Mapping, Sequence
```
Comment (FichteFoll) [inline st3/sublime_lib/_util/collection_util.py]: Rename to `collections.py` imo. "util" is already part of the module path.

File: st3/sublime_lib/_util/collection_util.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+
+__all__ = ['projection', 'get_selector', 'isiterable', 'ismapping', 'is_sequence_not_str']
+
+
+def projection(d, keys):
+    """
+    Return a new :class:`dict` with keys restricted to values in `keys`.
```
Comment (FichteFoll) [inline st3/sublime_lib/_util/collection_util.py]: should mention `d` here, e.g. ```keys of ``d`` ```.

Also ``` ``keys`` ```.

```suggestion
    Return a new :class:`dict` with keys of ``d`` restricted to values in ``keys``.
```

File: st3/sublime_lib/_util/collection_util.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+    .. code-block:: python
+
+       >>> projection({'a': 1, 'b': 2}, ['b'])
+       {'b': 2}
+
+    If `keys` is a :class:`dict`, then it maps keys of the original dict to
```
Comment (FichteFoll) [inline st3/sublime_lib/_util/collection_util.py]:
```suggestion
    If ``keys`` is a :class:`dict`, then it maps keys of the original dict to
```

File: st3/sublime_lib/_util/collection_util.py (hunk 1 above)
Comment (Thom1729) [reply st3/sublime_lib/_util/collection_util.py]: Done.

File: st3/sublime_lib/_util/collection_util.py (hunk 3 above)
Comment (Thom1729) [reply st3/sublime_lib/_util/collection_util.py]: Done.

File: st3/sublime_lib/_util/collection_util.py (hunk 2 above)
Comment (Thom1729) [reply st3/sublime_lib/_util/collection_util.py]: Done.

**Comment by Thom1729** [comment]
Addressed reviews and moved `glob_util`.

**Pull request merged**